### PR TITLE
[docker][containerd] Add open file descriptor metric

### PR DESCRIPTION
**Pull request by celenechang**
### What does this PR do?

Adds a new metric that tracks open file descriptors per container

### CODE REVIEW

File: releasenotes/notes/docker-open-file-descriptors-metric-49d5dfcfdb1b852f.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+enhancements:
+  - |
+    Extends the docker check to include an open file descriptors metric.
```
Comment (mfpierre) [inline releasenotes/notes/docker-open-file-descriptors-metric-49d5dfcfdb1b852f.yaml]: + containerd?

File: pkg/collector/corechecks/containers/containerd.go [hunk 2]
```diff
@@ -230,6 +231,25 @@ func computeMetrics(sender aggregator.Sender, cu cutil.ContainerdItf, fil *ddCon
...
+			pid := p.Pid
+			fdCount, err := cmetrics.GetFileDescriptorLen(int(pid))
+			if err != nil {
+				log.Warnf("Failed to get file desc length for pid %d, container %s: %s", pid, ctn.ID()[:12], err)
+				continue
+			}
+			fileDescCount += fdCount
+		}
+		sender.Gauge("containerd.proc.open_fds", float64(fileDescCount), "", tags)
+
```
Comment (mfpierre) [inline pkg/collector/corechecks/containers/containerd.go]: you can remove the extra line here

File: pkg/util/containers/metrics/proc_test.go [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+import (
+	"io/ioutil"
+	"strconv"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func setupFS(pidFdsMap map[int][]int) (*tempFolder, map[int]string, error) {
+
```
Comment (mfpierre) [inline pkg/util/containers/metrics/proc_test.go]: you can remove the extra line here

File: pkg/util/containers/metrics/proc_test.go [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+			if err != nil {
+				return nil, nil, err
+			}
+		}
+	}
+	return f, pidPathMap, nil
+}
+
+func TestGetFileDescriptorLen(t *testing.T) {
+
```
Comment (mfpierre) [inline pkg/util/containers/metrics/proc_test.go]: you can remove the extra line here

**Review by mfpierre — DISMISSED** [review]
Left a few nits but LGTM overall 🙌 

Don't forget to open up a PR on https://github.com/DataDog/integrations-core/blob/master/containerd/metadata.csv & https://github.com/DataDog/integrations-core/blob/master/docker_daemon/metadata.csv to add these new metrics metadata

**Comment by celenechang** [comment]
@mfpierre thanks for the review, and will do!

**Review by mfpierre — APPROVED** [review]

**Review by hush-hush — APPROVED** [review]

**Pull request merged**